src/react-components/home/HomePage.js

```diff
@@ -0,0 +1,100 @@
+import React, { useContext, useEffect } from "react";
+import { FormattedMessage, addLocaleData } from "react-intl";
+import en from "react-intl/locale-data/en";
+import classNames from "classnames";
+import configs from "../../utils/configs";
+import IfFeature from "../if-feature";
+import { Page } from "../layout/Page";
+import MediaTiles from "../media-tiles";
+import { CreateRoomButton } from "../input/CreateRoomButton";
+import { PWAButton } from "../input/PWAButton";
+import { useFeaturedRooms } from "./useFeaturedRooms";
+import styles from "./HomePage.scss";
+import mediaBrowserStyles from "../../assets/stylesheets/media-browser.scss";
+import discordLogoSmall from "../../assets/images/discord-logo-small.png";
+import { AuthContext } from "../auth/AuthContext";
+import { createAndRedirectToNewHub } from "../../utils/phoenix-utils";
+
+addLocaleData([...en]);
+
+export function HomePage() {
+  const featuredRooms = useFeaturedRooms();
+  const auth = useContext(AuthContext);
+
+  useEffect(() => {
+    const qs = new URLSearchParams(location.search);
+
+    // Support legacy sign in urls.
+    if (qs.has("sign_in")) {
+      const redirectUrl = new URL("/signin", window.location);
+      redirectUrl.search = location.search;
+      window.location = redirectUrl;
+    } else if (qs.has("auth_topic")) {
+      const redirectUrl = new URL("/verify", window.location);
```

Comment:
We will need to figure out the migration path for the emails we send out. Either we have to make a breaking change in reticulum (and make sure to notify everyone with a good lead up time so they can make sure their clients have been updated with the new endpoints) or just always support this version.

Reply:
I think we will need to make the breaking change in Reticulum, but keeping the old emails and this current redirect method for one release cycle seems like a good upgrade path.